packages/fx-core/src/core/middleware/utils/appYmlGenerator.ts

```diff
@@ -146,23 +147,10 @@ export class AppYmlGenerator extends BaseAppYmlGenerator {
 
   private async generateAzureHandlebarsContext(): Promise<void> {
     // isFunctionBot
-    const pluginSettings = this.oldProjectSettings.pluginSettings;
-    if (
-      pluginSettings &&
-      pluginSettings["fx-resource-bot"] &&
-      pluginSettings["fx-resource-bot"]["host-type"] === "azure-function"
-    ) {
-      this.handlebarsContext.isFunctionBot = true;
-    }
-    // isWebAppBot and the resourceId in bicep should be "botWebAppResourceId", then map state.fx-resource-bot.botWebAppResourceId
-    if (
-      pluginSettings &&
-      pluginSettings["fx-resource-bot"] &&
-      pluginSettings["fx-resource-bot"]["host-type"] === "app-service" &&
-      this.bicepContent.includes("botWebAppResourceId")
-    ) {
-      this.handlebarsContext.isWebAppBot = true;
-    }
+    const projectSettings: ProjectSettingsV3 = this.oldProjectSettings;
+    this.handlebarsContext.isFunctionBot = hasFunctionBot(projectSettings);
+    this.handlebarsContext.isWebAppBot =
```

Comment:
Below code could be clearer. Otherwise people may be hard to understand why we need to read bicep content to determine whether it's a web app bot.
```
this.handlebarsContext.isWebAppBot = hasWebAppBot(projectSettings);
this.handlebarsContext.useBotWebAppResourceId = this.bicepContent.includes("botWebAppResourceId");
```
PS: Also needs to update the yaml template

Reply:
modified